test/lambda_toolshed/papillon_test.cljc

```diff
@@ -224,3 +258,24 @@
           (is (empty? (::ix/queue res)))
           (is (empty? (::ix/stack res)))
           (is (= expected-log (::ix/trace res))))))))
+
+#?(:cljs
+   (deftest allows-for-promise-rejection-return-values
+     (let [the-exception (ex-info "the exception" {})
+           ixs [{:name :ix}
+                capture-ix
+                {:name :promise-rejector :enter (fn [_] (js/Promise.reject the-exception))}]
+           expected-log [[:ix :enter]
+                         [:capture :enter]
+                         [:promise-rejector :enter]
+                         [:promise-rejector :error]
+                         [:capture :error]
+                         [:ix :leave]]]
+       (go-test
+        (let [[res _] (alts! [(ix/execute ixs {::ix/trace []})
+                              (async/timeout 10)])]
+          (is (map? res))
+          (is (empty? (::ix/queue res)))
+          (is (empty? (::ix/stack res)))
+          (is (= (ex-message the-exception) (ex-message (res ::error))))
+          (is (= expected-log (::ix/trace res))))))))
```

Comment:
After some feedback on how it should behave, I will add some tests for the 4-arity version.